### PR TITLE
Fix multi-join dynamic filtering

### PR DESCRIPTION
**Pull request by kewang1024**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java [hunk 1]
```diff
@@ -123,6 +125,9 @@ public PlanWithConsumedDynamicFilters visitJoin(JoinNode node, Set<String> allow
         {
             JoinDynamicFilterResult joinDynamicFilterResult = extractDynamicFilterFromJoin(node, allowedDynamicFilterIds);
             if (!joinDynamicFilterResult.getProbe().equals(node.getLeft()) || !joinDynamicFilterResult.getBuild().equals(node.getRight()) || !joinDynamicFilterResult.getDynamicFilters().equals(node.getDynamicFilters())) {
+                Optional<RowExpression> filter = node
+                        .getFilter().map(this::removeAllDynamicFilters)  // dynamic filtering is not supported for LookupJoinOperators
+                        .filter(expression -> !expression.equals(TRUE_LITERAL));
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java:130]: `TRUE_CONSTANT`. I assume this is cherry picked from prestosql? Do you have a reference?

Comment (kewang1024) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java:130]: https://github.com/prestosql/presto/pull/2659

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/DynamicFiltersChecker.java [hunk 2]
```diff
@@ -86,14 +94,14 @@ public void validate(PlanNode plan, Session session, Metadata metadata, SqlParse
             {
                 Set<String> currentJoinDynamicFilters = node.getDynamicFilters().keySet();
                 Set<String> consumedProbeSide = node.getProbe().accept(this, context);
-                verify(
-                        difference(currentJoinDynamicFilters, consumedProbeSide).isEmpty(),
-                        "Dynamic filters present in join were not fully consumed by its probe side.");
+                Set<String> unconsumedByProbeSide = difference(currentJoinDynamicFilters, consumedProbeSide);
+                verify(unconsumedByProbeSide.isEmpty(),
+                        "Dynamic filters %s present in join were not fully consumed by its probe side.", unconsumedByProbeSide);
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/DynamicFiltersChecker.java:99]: same line or one param per line

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/DynamicFiltersChecker.java [hunk 3]
```diff
@@ -86,14 +94,14 @@ public void validate(PlanNode plan, Session session, Metadata metadata, SqlParse
...
 
                 Set<String> consumedBuildSide = node.getBuild().accept(this, context);
-                verify(
-                        intersection(currentJoinDynamicFilters, consumedBuildSide).isEmpty(),
-                        "Dynamic filters present in join were consumed by its build side.");
+                Set<String> unconsumedByBuildSide = intersection(currentJoinDynamicFilters, consumedBuildSide);
+                verify(unconsumedByBuildSide.isEmpty(),
+                        "Dynamic filters %s present in join were consumed by its build side.", unconsumedByBuildSide);
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/DynamicFiltersChecker.java:104]: same line or one param per line

**Pull request merged**